main.go

```diff
@@ -307,42 +333,52 @@ func deploy(clearedFilesToDeploy []string, config Config) (map[string]string, er
+			artifactURLs, err := uploaders.DeployIPA(pth, config.BuildURL, config.APIToken, config.NotifyUserGroups, config.NotifyEmailList, config.IsPublicPageEnabled)
 			if err != nil {
-				return nil, fmt.Errorf("deploy failed, error: %s", err)
+				return ArtifactURLCollection{}, fmt.Errorf("deploy failed, error: %s", err)
 			}
 
-			if installPage != "" {
-				publicInstallPages[filepath.Base(pth)] = installPage
-			}
+			fillURLMaps(artifactURLCollection, artifactURLs, pth, isPublic)
 		case ".aab":
 			log.Donef("Uploading aab file: %s", pth)
 
-			if err := uploaders.DeployAAB(pth, androidArtifacts, config.BuildURL, config.APIToken,
-				config.NotifyUserGroups, config.NotifyEmailList, config.IsPublicPageEnabled, config.BundletoolVersion); err != nil {
-				return nil, fmt.Errorf("deploy failed, error: %s", err)
+			artifactURLs, err := uploaders.DeployAAB(pth, androidArtifacts, config.BuildURL, config.APIToken, config.BundletoolVersion)
+			if err != nil {
+				return ArtifactURLCollection{}, fmt.Errorf("deploy failed, error: %s", err)
 			}
+
+			fillURLMaps(artifactURLCollection, artifactURLs, pth, isPublic)
 		case zippedXcarchiveExt:
 			log.Donef("Uploading xcarchive file: %s", pth)
-			if err := uploaders.DeployXcarchive(pth, config.BuildURL, config.APIToken); err != nil {
-				return nil, fmt.Errorf("deploy failed, error: %s", err)
+
+			artifactURLs, err := uploaders.DeployXcarchive(pth, config.BuildURL, config.APIToken)
+			if err != nil {
+				return ArtifactURLCollection{}, fmt.Errorf("deploy failed, error: %s", err)
 			}
+			fillURLMaps(artifactURLCollection, artifactURLs, pth, isPublic)
 		default:
 			log.Donef("Uploading file: %s", pth)
 
-			installPage, err := uploaders.DeployFile(pth, config.BuildURL, config.APIToken, config.NotifyUserGroups, config.NotifyEmailList, config.IsPublicPageEnabled)
+			artifactURLs, err := uploaders.DeployFile(pth, config.BuildURL, config.APIToken)
 			if err != nil {
-				return nil, fmt.Errorf("deploy failed, error: %s", err)
+				return ArtifactURLCollection{}, fmt.Errorf("deploy failed, error: %s", err)
 			}
 
-			if installPage != "" {
-				publicInstallPages[filepath.Base(pth)] = installPage
-			} else if config.IsPublicPageEnabled == "true" {
-				log.Warnf("is_enable_public_page is set, but public download isn't allowed for this type of file")
-			}
+			fillURLMaps(artifactURLCollection, artifactURLs, pth, isPublic)
 		}
 	}
-	return publicInstallPages, nil
+	return artifactURLCollection, nil
+}
+
+func fillURLMaps(artifactURLCollection ArtifactURLCollection, artifactURLs uploaders.ArtifactURLs, apk string, isPublic bool) {
+	if artifactURLs.PublicInstallPageURL != "" {
+		artifactURLCollection.PublicInstallPageURLs[filepath.Base(apk)] = artifactURLs.PublicInstallPageURL
+	} else if isPublic {
+		log.Warnf("is_enable_public_page is set, but public download isn't allowed for this type of file")
```

Comment:
I am not sure if moving this warning here makes sense, previously we printed this only in case of fileupload, now we handle it for each extension.

I think the previous version was not fine either, as the user did not misconfigure anything, but we still print a warning. IMO we should explain for what file the step will generate a public install page and remove this warning message.

Reply:
previously we filled the `publicInstallPages` only in case of certain file types, now we fill it based on if the file has got the URL or not. Did not we change here what gets into the `publicInstallPages `? Like a `.aab` will not get a public install page?

Reply:
I wanted to change the text of the warning, thanks :)

Reply:
Yes you're right I'm gonna add an additional flag